src/languages/delphi.js

```diff
@@ -164,20 +164,28 @@ export default function(hljs) {
     relevance: 0,
     // Source: https://www.freepascal.org/docs-html/ref/refse6.html
     variants: [
+      {
+        // Regular numbers, e.g., 123, 123.456.
+        begin: /\b\d[\d_]*(\.\d[\d_]*)?/ },
       {
         // Hexadecimal notation, e.g., $7F.
-        begin: '\\$[0-9A-Fa-f]+' },
+        begin: /\$[\dA-Fa-f_]*/ },
```

Comment:
This is not comparable because it can match no digits at all.

Reply:
Hi @joshgoebel,
This was an intentional change in [5d3110c](https://github.com/highlightjs/highlight.js/pull/3932/commits/5d3110c91f0869825f584d6645383de8a34707b1).
Delphi allows all digits to be omitted in hexadecimal literals.

Reply:
What is the purpose and value of a hexadecimal literal with no digits?

Reply:
> What is the purpose

From a language design perspective, I agree that it's a pretty bizarre thing for Delphi to allow.
My change was purely in the interest of accuracy to Delphi's official parser.

> and value

The value of the expression (in both cases) is 0.

Reply:
I'm worried auto-detect and other languages... $ and % are weird things to match on their own (and things that occur often in many types of files)... if we're going to add them now I think we need the original rule PLUS a new rule that matches `*` (set to 0 relevance) - so as not to influence auto-detection...

Or we can hold this until v12 when we stop caring about such things.

Reply:
> if we're going to add them now I think we need the original rule PLUS a new rule that matches `*` (set to 0 relevance) - so as not to influence auto-detection...

Very sensible! I'm happy to make that change.

Reply:
Done.